src/ApplicationUI/Forms/Buttons/Button.vue

```diff
@@ -0,0 +1,177 @@
+<template>
+  <Component :is="as" :disabled="disabled || loading" :type="type" :class="classes">
+    <SpinnerIcon v-if="loading" :class="iconClasses"/>
+    <span v-else :class="iconClasses">
+      <slot name="icon"/>
+    </span>
+    <span v-if="loading">{{ loadingText }}</span>
+    <slot v-else/>
+  </Component>
+</template>
+
+<script setup>
+import SpinnerIcon from '@/ApplicationUI/Icons/SpinnerIcon.vue';
+import { computed, ref } from 'vue';
+import { get } from '@vueuse/core';
+
+const props = defineProps({
+  as: {
+    type: [String, Object],
+    default: () => 'button',
+  },
+  type: {
+    type: String,
+    default: undefined,
+  },
+  noFlex: {
+    type: Boolean,
+    default: false,
+  },
+  noPadding: {
+    type: Boolean,
+    default: false,
+  },
+  noBorder: {
+    type: Boolean,
+    default: false,
+  },
+  noShadow: {
+    type: Boolean,
+    default: false,
+  },
+  noRound: {
+    type: Boolean,
+    default: false,
+  },
+  noRing: {
+    type: Boolean,
+    default: false,
+  },
+  theme: {
+    type: String,
+    default: 'default',
+    validator: (value) => ['default', 'primary', 'danger', 'cancel'].includes(value),
```

Comment:
`reset` instead of `cancel` as name?

Reply:
IMO `reset` is a sort of implementation of a more generic `cancel` - for instance a `cancel` button could be used to dismiss a modal or something like this, and would have the same design as a (form) `reset` button.

Reply:
I agree for this repository :+1: